backend/src/bin/export_schema.rs

```diff
@@ -2,14 +2,27 @@
 //! code. The schema is required for the frontend to compile.
 
 use anyhow::Result;
+use std::{fmt, path::Path};
 
 #[path = "../api/mod.rs"]
 mod api;
 
+#[derive(Debug)]
+struct InvalidPath;
+impl fmt::Display for InvalidPath {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "invalid path")
+    }
+}
+impl std::error::Error for InvalidPath {}
+
 fn main() -> Result<()> {
     let schema = api::root_node().as_schema_language();
 
     if let Some(target) = std::env::args().nth(1) {
+        std::fs::create_dir_all(Path::new(
+            Path::new(&target).parent().ok_or(InvalidPath)?,
+        ))?;
```

Comment:
Somehow I was under the impression that `parent` returned a string of some sort. Whoopsy. :shrug: 

---

I guess making the "parent creation" conditional is indeed nicer than the whole error shenanigans, but do note that it would still execute for "local paths" like `schema.graphql`; for those, `parent` returns `Some("")`: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=97bca31cd4f9e35234e01904e21b7318